src/prefect/blocks/core.py

```diff
@@ -950,11 +950,14 @@ async def _save(
                 `is_anonymous` is `True`.
         """
         if name is None and not is_anonymous:
-            raise ValueError(
-                "You're attempting to save a block document without a name. "
-                "Please either save a block document with a name or set "
-                "is_anonymous to True."
-            )
+            if self._block_document_name is None:
+                raise ValueError(
+                    "You're attempting to save a block document without a name. "
+                    "Please either save a block document with a name or set "
```

Comment:
nit: I'm wondering how many users are aware of what a block document is, rather than just knowing it as a block.